lib/common/camera/3rd-Person.lua

```diff
@@ -0,0 +1,240 @@
+		* CFrame.new(0, 1, 0)
+end
+
+-- Returns goal CFrame when there is no target
+local function RenderNoTarget()
+	
+	-- Very slowly rotates over
+	return camera.CFrame * CFrame.Angles(0, 0, 0.01)
+end
+
+-- Function that adds any relevant blur or FoV changes
+local function PostProcessing(cameraState)
+	
+	local blurDistance = 0
+	
+	
+	-- If there is no target, rapidly increase blur
+	if(cameraState == cameraStates.NO_TARGET)then
+		
+		blurDistance = blur.Size + 5
+		
+	-- If the camera is pulling out, there should be a normal blur increase	
+	elseif(cameraState == cameraStates.PULLBACK)then
+		
+		blurDistance = blur.Size + (camera.CFrame.p - target.Position).magnitude / 5
+		
+	else
+		
+		-- Else, slow blur increase
+		blurDistance = (camera.CFrame.p - target.Position).magnitude / 10
+	end
+	
+	-- If it's close enough, let's just ensure there's no blur
+	if(blurDistance < 5)then
+		blurDistance = 0
+	end
+	
+	-- Create and play this tween
+	tweenService:Create(blur, tweenInfo, {Size = blurDistance}):Play()
+	
+	-- Grabs the speed of the target, and dictates goalFoV depending on ceiling and floor as configured
+	local targetSpeed = target.Velocity.magnitude
+	local goalFoV = math.max(math.min(targetSpeed, ceilFoV), floorFoV)
+	
+	-- Create and play the FoV tween
+	tweenService:Create(camera, tweenInfo, {FieldOfView = ((goalFoV - camera.FieldOfView)/10) + camera.FieldOfView}):Play()
+	
+end
+
+-- Function that controls the camera
+function Render()
+	-- Waits camera speed rate
+	wait(cameraSpeed)
+	
+	-- Ensures the camera focus is correct
+	camera.Focus = target.CFrame
+	
+	-- Gets current states
+	local humanoidState = humanoid:GetState()
+	local cameraState = GetCameraState(humanoidState)
+	
+	-- Setup CFrames
+	local currentCFrame = camera.CFrame
+	local targetCFrame = nil
+	
+	-- Calls the correct method depending on the camera state
+	if(cameraState == cameraStates.TRACK)then
+		
+		targetCFrame = RenderTrack()
+		
+	elseif(cameraState == cameraStates.NO_TRACK)then
+
+		targetCFrame = RenderNoTrack(currentCFrame)
+	
+	elseif(cameraState == cameraStates.PULLBACK)then
+		
+		targetCFrame = RenderPullback(currentCFrame)
+			
+	elseif(cameraState == cameraStates.NO_TARGET)then
+		
+		targetCFrame = RenderNoTarget()
+	end
+	
+	-- Needs to update incase the cameraState manually changed the CFrame
+	currentCFrame = camera.CFrame
+	
+	-- If the player is very close to the camera, it should react slightly faster
+	local tempReaction = cameraReaction
+	if((currentCFrame.p - target.Position).magnitude < 8)then
+		tempReaction = tempReaction * 2
+		
+	end
+	
+	local newCFrame = currentCFrame:lerp(targetCFrame, tempReaction)
+	
+	-- Create and play the tween
+	cameraCFrameTween = tweenService:Create(camera, tweenInfo, {CFrame = newCFrame})
+	cameraCFrameTween:Play()
+	
+	-- Generate any necessary post processing
+	PostProcessing(cameraState)
+end
+
+-- When the player respawns, need to update this
+player.CharacterAdded:Connect(function()
```

Comment:
No, because it is remade every time the character resets, therefore not allowing anything to occur within the respawn time as far as the camera is concerned.